internal/services/servicenetworking/application_load_balancer_association_resource.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type AssociationResource struct{}
+
+type AssociationModel struct {
+	Name                      string            `tfschema:"name"`
+	ApplicationLoadBalancerId string            `tfschema:"application_load_balancer_id"`
+	SubnetId                  string            `tfschema:"subnet_id"`
+	Tags                      map[string]string `tfschema:"tags"`
+}
+
+var _ sdk.ResourceWithUpdate = AssociationResource{}
+
+func (t AssociationResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"application_load_balancer_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: associationsinterface.ValidateTrafficControllerID,
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (t AssociationResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (t AssociationResource) ModelObject() interface{} {
+	return &AssociationModel{}
+}
+
+func (t AssociationResource) ResourceType() string {
+	return "azurerm_application_load_balancer_association"
+}
+
+func (t AssociationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return associationsinterface.ValidateAssociationID
+}
+func (t AssociationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			trafficControllerClient := metadata.Client.ServiceNetworking.TrafficControllerInterface
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			var config AssociationModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			parsedTrafficControllerId, err := associationsinterface.ParseTrafficControllerID(config.ApplicationLoadBalancerId)
+			if err != nil {
+				return err
+			}
+
+			controllerId := trafficcontrollerinterface.NewTrafficControllerID(parsedTrafficControllerId.SubscriptionId, parsedTrafficControllerId.ResourceGroupName, parsedTrafficControllerId.TrafficControllerName)
+			controller, err := trafficControllerClient.Get(ctx, controllerId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", controllerId, err)
+			}
+
+			if controller.Model == nil {
+				return fmt.Errorf("retrieving %s: Model was nil", controllerId)
+			}
+
+			loc := controller.Model.Location
```

Comment:
This prefers to be changed into below (See [this](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md#:~:text=//%20the%20SDK%20will,from%0A%09%09%09//%20crashing.) for the why):

```suggestion
			var loc string
			if controller.Model != nil {
				loc = controller.Model.Location
			}
```